src/ldp/http/metadata/ContentTypeParser.ts

```diff
@@ -0,0 +1,17 @@
+import type { HttpRequest } from '../../../server/HttpRequest';
+import type { RepresentationMetadata } from '../../representation/RepresentationMetadata';
+import type { MetadataParser } from './MetadataParser';
+
+/**
+ * Parser for the `content-type` header.
+ * Currently only stores the media type and ignores other parameters such as charset.
+ */
+export class ContentTypeParser implements MetadataParser {
+  public async parse(request: HttpRequest, metadata: RepresentationMetadata): Promise<void> {
+    const contentType = request.headers['content-type'];
+    if (contentType) {
+      // Will need to use HeaderUtil once parameters need to be parsed
+      metadata.contentType = /^[^;]*/u.exec(contentType)![0].trim();
+    }
```

Comment:
```suggestion
    const contentType = request.headers['content-type'] || '';
    // Will need to use HeaderUtil once parameters need to be parsed
    metadata.contentType = /^[^; ]*/u.exec(contentType)![0];
```

Reply:
This would set the contentType to an empty string instead of undefined though.

Reply:
Yeah, never mind indeed.